src/web/rustdoc.rs

```diff
@@ -210,7 +213,12 @@ pub(crate) async fn rustdoc_redirector_handler(
                     {
                         debug!(?target, ?err, "got error serving file");
                     }
-                    return Err(AxumNope::ResourceNotFound);
+                    // FIXME: we sometimes still get requests for toolchain
+                    // specific static assets under the crate/version/ path.
+                    // This is fixed in rustdoc, but pending a rebuild for
+                    // docs that were affected by this bug.
+                    // https://github.com/rust-lang/docs.rs/issues/1979
+                    return try_serve_legacy_toolchain_asset(storage, config, target).await;
```

Comment:
To avoid letting another file regress we could check that the filename starts with `search-2021` or `search-2022`.

Reply:
I updated the code, checking for a `search-` prefix